### ISSUE TITLE
workflow of finetune when accuracy loss occur

### ISSUE THREAD

**Issue by erwangccc** [VainF/Torch-Pruning#213]
Hi, @VainF 

Thanks for your great work first!
It will always loss acc after we pruned a model under my understanding, so we need finetuning our model after pruning.

So the workflow of finetune is as following?

```
# set progressive pruning step
iterative_steps = 1 
# set pruner
pruner = tp.pruner.MagnitudePruner(
    model,
    example_inputs,
    importance=imp,
    iterative_steps=iterative_steps,
    ch_sparsity=0.5, # remove 50% channels, ResNet18 = {64, 128, 256, 512} => ResNet18_Half = {32, 64, 128, 256}
    ignored_layers=ignored_layers,
    unwrapped_parameters=unwrapped_parameters
)
# call pruner.step() to prune the model
pruner.step()
print(model)

......

# train the model
for epoch in range(start_epoch, epochs):  # epoch ------------------------------------------------------------------
        model.train()
```

Is it correct?

**Comment by VainF** [comment]
Yeh, it's correct!

**Comment by erwangccc** [comment]
@VainF 

Thanks. by the way, have you compared method such as training model while pruning and training after pruning, which one will be better?

**Comment by AymenBOUGUERRA** [comment]
@erwangccc 

> It will always loss acc after we pruned a model under my understanding, so we need finetuning our model after pruning.

Using a sparsity ratio of 0.5 will end up irreversibly destroying your feature maps, the evaluation score of the pruned model right after the pruning step is 0, what you will need is to not only fine tune the model but rather retrain it from scratch (one advantage is that you will be able to double your batch-size due to the pruning which will speed up the process and give you better BatchNorms)

> Thanks. by the way, have you compared method such as training model while pruning and training after pruning, which one will be better?

I have also used the author's script to implement gradual pruning on Yolov7, and there were absolutely no benefit doing so, it only took longer to converge as YOLOs are mainly constituted of Conv2d layers, this method would be useful for Dense layers.

**Conclusion**: it is faster to use a sparsity ratio such as 0.5 and then retrain the model from scratch then gradually pruning the model by 0.05 10 times and **fine-tuning*** for X epochs between each pruning step (in the case of Conv models).
*using a small sparsity ratio will maintain a big portion of the information in the feature maps.
